Update Balance screen

erdemyerebasmaz opened the pull request:
This PR partially addresses
- #604

#### Changelist:
- Converted `WalletDashboard` and it's child widgets to `StatefulWidget's`.

#### Notes:
@roeierez Haven't debugged it yet but my initial impression is this is a asynchronous communication issue regarding calling `sync` on **SDK** side(_either on a payment sent/received flow or on receiving a `BreezEvent` on `BreezBridge`_).

roeierez commented:
@erdemyerebasmaz From my research this is not the case as the sdk only returns from payment after it syncs.
Also found an issue in greenlight regarding that and is under Christian's investigation.

erdemyerebasmaz commented:
@roeierez Thanks, saw your discussion now. Either way, this refactoring is a plus and is more alike to breezmobile.

- Converted WalletDashboard and it's child widgets to StatefulWidget's.
- All child widgets use Bloc states from a single source,
- Extracted fiat balance text into it's own widget.

ubbabeck reviewed (approved):
Looks good! Nice refactor 👍 
Tested receive and fiat currency flow on an android emulator, did not notice any issues.
